internal/client.go

```diff
@@ -626,6 +630,515 @@ type (
+		GetID() string
+
+		// Delete the Schedul
+		Delete(ctx context.Context) error
+
+		// Update the Schedule. fn takes a description of the schedule and returns the new desired schedule.
+		// If update returns a nil response then no update will occur.
+		//
+		// NOTE: If two Update calls are made in parallel to the same Schedule there is the potential
+		// for a race condition.
+		Update(ctx context.Context, fn func(*ScheduleDescribeResponse) *ScheduleUpdate, options ScheduleUpdateOptions) error
+
+		// Describe fetches the Schedule's description from the Server
+		Describe(ctx context.Context) (*ScheduleDescribeResponse, error)
+
+		// Trigger an Action to be taken immediately. Will override the schedules default policy
+		// with the one specified here.
+		Trigger(ctx context.Context, overlap enumspb.ScheduleOverlapPolicy) error
+
+		// Pause the Schedule will also overwrite the Schedules current note with the new note.
+		// If an empty note is passed the Schedules note will be set to 'Paused via Go SDK'.
+		Pause(ctx context.Context, note string) error
+
+		// Unpause the Schedule will also overwrite the Schedules current note with the new note.
+		// If an empty note is passed the Schedules note will be set to 'Unpaused via Go SDK'.
+		Unpause(ctx context.Context, note string) error
+	}
+
+	// ScheduleActionResult describes when a schedule action took place
+	ScheduleActionResult struct {
+		// ScheduleTime - Time that the Action was scheduled for, including jitter.
+		ScheduleTime time.Time
+
+		// ActualTime - Time that the Action was actually taken.
+		ActualTime time.Time
+
+		// StartWorkflowResult - If action was ScheduleWorkflowAction, returns the
+		// ID of the workflow.
+		StartWorkflowResult ScheduleWorkflowExecution
+	}
+
+	// ScheduleListEntry
+	ScheduleListEntry struct {
+		// ScheduleID - The business identifier of the schedule.
+		ScheduleID string
+
+		// Spec - Describes when Actions should be taken.
+		Spec *ScheduleSpec
+
+		// Note - Informative human-readable message with contextual notes, e.g. the reason
+		// a Schedule is paused. The system may overwrite this message on certain
+		// conditions, e.g. when pause-on-failure happens.
+		Note string
+
+		// Paused - True if the schedule is paused.
+		Paused bool
+
+		// WorkflowType - If the schedule action is a Wokrflow then
+		// describes what workflow is run.
+		WorkflowType WorkflowType
+
+		// RecentActions- Most recent 10 Actions started (including manual triggers).
+		//
+		// Sorted from older start time to newer.
+		RecentActions []ScheduleActionResult
+
+		// NextActionTimes - Next 10 scheduled Action times.
+		NextActionTimes []time.Time
+
+		// Memo - Non-indexed user supplied information.
+		Memo *commonpb.Memo
+
+		// SearchAttributes - Indexed info that can be used in query of List schedules APIs (only
+		// supported when Temporal server is using ElasticSearch). The key and value type must be registered on Temporal server side.
```

Comment:
we probably shouldn't mention Elasticsearch in the comments here. we call it the feature set that includes search attributes "Advanced Visibility" elsewhere. also, it will soon be supported for all backends

Reply:
Yeah I took the wording from `StartWorkflowOptions` but standardizing on "Advanced Visibility" is better, I think.